packages/data/addon/utils/classes/cache.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2022, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+type CacheRecord<T> = {
+  creationTime: number; // the time this record was created
+  lastAccessed: number; // the last time this record was accessed
+  records: T; // the actual records
+};
+
+/**
+ * a cache
+ * to use, give it a unique string (cacheId) and whatever you want stored alongside that key
+ * returns only exact matches
+ */
+export default class Cache<T> {
+  private _maxCacheSize;
```

Comment:
Preference here, but I think caches should have a `Map` like interface. (with a few extra methods for cache stuff). So my suggestions will be based in that preference.